gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/GaaSJobObservabilityEventProducer.java

```diff
@@ -123,7 +129,11 @@ private void setupMetrics(State state) {
 
   public void emitObservabilityEvent(final State jobState) {
     GaaSJobObservabilityEvent event = createGaaSObservabilityEvent(jobState);
-    sendUnderlyingEvent(event);
+    if (jobState.getProp(TimingEvent.FlowEventConstants.JOB_NAME_FIELD).equals(JobStatusRetriever.NA_KEY) && this.emitFlowObservabilityEvent) {
```

Comment:
why do we only emit flow type events in some cases? is this for backwards compatibility? what's the harm in emitting FlowLevel events going forward?

Reply:
Generally for backwards compatibility in every event producer, easier to make it opt-in as its a change in semantics. 
With flow level events, any job level event that has "NA" as the jobName would be omitted now and put under a different topic name (if done with Kafka)

Reply:
will it throw NPE if jobState does not have job_name field? `jobState.getProp(TimingEvent.FlowEventConstants.JOB_NAME_FIELD).equals(JobStatusRetriever.NA_KEY`.

Reply:
At this point JOB_NAME_FIELD gets populated by the jobstatus monitor before it emits the observability event
```
  static Pair<org.apache.gobblin.configuration.State, NewState> recalcJobStatus(org.apache.gobblin.configuration.State jobStatus,
      StateStore<org.apache.gobblin.configuration.State> stateStore) throws IOException {
    try {
      if (!jobStatus.contains(TimingEvent.FlowEventConstants.JOB_NAME_FIELD)) {
        jobStatus.setProp(TimingEvent.FlowEventConstants.JOB_NAME_FIELD, JobStatusRetriever.NA_KEY);
      }
```      
and this is called right after parsing the job status